packages/flutter_tools/lib/src/flutter_cache.dart

```diff
@@ -235,18 +235,21 @@ class FlutterSdk extends EngineCachedArtifact {
 
   @override
   List<List<String>> getBinaryDirs() {
-    // Currently only Linux supports both arm64 and x64.
+    // Linux and Windows both support arm64 and x64.
     final String arch = cache.getHostPlatformArchName();
     return <List<String>>[
       <String>['common', 'flutter_patched_sdk.zip'],
       <String>['common', 'flutter_patched_sdk_product.zip'],
       if (cache.includeAllPlatforms) ...<List<String>>[
         <String>['windows-x64', 'windows-x64/artifacts.zip'],
+        <String>['windows-arm64', 'windows-arm64/artifacts.zip'],
         <String>['linux-$arch', 'linux-$arch/artifacts.zip'],
         <String>['darwin-x64', 'darwin-$arch/artifacts.zip'],
       ]
-      else if (_platform.isWindows)
-        <String>['windows-x64', 'windows-x64/artifacts.zip']
+      else if (_platform.isWindows) ...<List<String>>[
+        <String>['windows-x64', 'windows-x64/artifacts.zip'],
+        <String>['windows-arm64', 'windows-arm64/artifacts.zip'],
+      ]
```

Comment:
Self-note to follow-up: Windows engine artifacts are huge. Ideally we only download the artifacts necessary for the target platform. If we only target Windows Arm64, we shouldn't download Windows x64 artifacts. Double-check this.

Reply:
Updated this PR to only do native compilation. Thus, we only download native artifacts, and there is no need to have both architectures.